### PR TITLE
added nano ble sense rev2 firmware

### PR DESCRIPTION
**Pull request by jhansson-ard**
Added the firmware for the Nano BLE Sense Rev2 for connection with the device and the board through BLE

### CODE REVIEW

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+void loop() {
+  BLE.poll(1000);
+  while (BLE.connected()) {
+    lowPowerBleWait(100);
+    updateSubscribedCharacteristics();
+  }
+}
+
+void updateSubscribedCharacteristics() {
+  Serial.println("updateSubscribedCharacteristics");
```
Comment (Rocketct) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino:163]: remove serial print

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+  if (accelerationCharacteristic.subscribed()) {
+    Serial.println("subricption done");
```
Comment (Rocketct) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino:165]: remove serial print

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    if (IMU.accelerationAvailable()) {
+      Serial.println("acceleration data available");
```
Comment (Rocketct) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino:167]: same as above

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+      acceleration[0] = x;
+      acceleration[1] = y;
+      acceleration[2] = z;
+      Serial.print("values: ");
+      Serial.print(x);
+      Serial.print(" ");
+      Serial.print(y);
+      Serial.print(" ");
+      Serial.println(z);
```
Comment (Rocketct) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino:180]: same as above

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/config.h [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+const int FlashChipSelect = 2;
+
+#include <Arduino_APDS9960.h>
+#include <Arduino_HS300x.h> // hs3003
+#include <Arduino_LPS22HB.h>
+
+// IMU
+// BMM150 and bmi 270
+//#include <BoschSensorClass.h>
```
Comment (Rocketct) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/config.h:15]: erase the commented include **#include <BoschSensorClass.h>**

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/config.h [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  while (1) {
+    digitalWrite(LED_BUILTIN, HIGH);
+    delay(500);
+    digitalWrite(LED_BUILTIN, LOW);
+    delay(500);
+  }
+}
+
+void sensorsInit() {
+  // bmm150
```
Comment (Rocketct) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/config.h:41]: // bmm150 -> // bmm150 and bmi270

**Review by Rocketct — CHANGES_REQUESTED** [review]

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino (hunk 1 above)
Comment (jhansson-ard) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino:163]:
```suggestion
```

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino (hunk 2 above)
Comment (jhansson-ard) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino:165]:
```suggestion
```

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
```
Comment (jhansson-ard) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino:178]:
```suggestion
```

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+  while (BLE.connected()) {
+    lowPowerBleWait(100);
+    updateSubscribedCharacteristics();
+  }
+}
+
+void updateSubscribedCharacteristics() {
+  if (accelerationCharacteristic.subscribed()) {
+    if (IMU.accelerationAvailable()) {
+      Serial.println("acceleration data available");
```
Comment (jhansson-ard) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino:165]:
```suggestion
```

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/config.h (hunk 5 above)
Comment (jhansson-ard) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/config.h:15]:
```suggestion
```

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/config.h (hunk 6 above)
Comment (jhansson-ard) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/config.h:41]:
```suggestion
  // bmm150 and bmi270
```

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+  service.addCharacteristic(soundPressureCharacteristic);
+  service.addCharacteristic(resistanceCharacteristic);
+
+  versionCharacteristic.setValue(VERSION);
+
+  BLE.addService(service);
+  BLE.advertise();
+
+  lowPower();
+  Serial.println("Initialization done!");
```
Comment (Rocketct) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino:151]: remove this serial print too

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/config.h [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+// IMU
+// BMM150 and bmi 270
+#include "Arduino_BMI270_BMM150.h"
+
+
+#include <avr/dtostrf.h>
+
+
+//BoschSensorClass BME = BoschSensorClass(Wire);
```
Comment (Rocketct) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/config.h:21]: remove this comment

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+    Serial.println(msg);
+  }
+#endif
+}
+
+
+
+void setup() {
+#ifdef DEBUG
+  Serial.begin(115200);
```
Comment (Rocketct) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino:85]: change to 9600

**Review by Rocketct — CHANGES_REQUESTED** [review]

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino (hunk 11 above)
Comment (jhansson-ard) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino:85]:
```suggestion
  Serial.begin(9600);
```

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino (hunk 9 above)
Comment (jhansson-ard) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/Nano33BLESenseRev2Firmware.ino:151]:
```suggestion
```

File: examples/ScienceKit/Nano33BLESenseRev2Firmware/config.h (hunk 10 above)
Comment (jhansson-ard) [inline examples/ScienceKit/Nano33BLESenseRev2Firmware/config.h:21]:
```suggestion
```

**Review by Rocketct — APPROVED** [review]

**Pull request merged**